dask/array/rechunk.py

```diff
@@ -220,9 +226,20 @@ def rechunk(x, chunks="auto", threshold=None, block_size_limit=None):
 
     >>> y = x.rechunk({0: -1, 1: 'auto'}, block_size_limit=1e8)
     """
+    if n_chunks:
+        if isinstance(n_chunks, (int, float)):
+            n_chunks = (n_chunks,) * x.ndim
```

Comment:
I wonder if this behavior might surprise people in the tall-and-skinny case common in ML workflows.  I don't really have a better suggestion here, other than to force the user to provide a tuple, but I thought I'd bring it up.

Reply:
> force the user to provide a tuple,

I like that better. That's I'm using in example code.